tests/ci/benchmark_framework/compare_results.py

```diff
@@ -0,0 +1,74 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import pandas as pd
+import numpy as np
+
+
+# helper function to read json or csv file data obtained from the speed tool into a pandas dataframe
+def read_data(file):
+    if file.endswith(".json"):
+        df = pd.read_json(file)
+    else:
+        # this assumes we're using a csv generated by convert_json_to_csv.py
+        df = pd.read_csv(file, skiprows=1, index_col=0)
+    return df
+
+
+if len(sys.argv) != 4:
+    print("Usage: compare_results.py [file1] [file2] [output filename]", file=sys.stderr)
+    sys.exit(1)
+
+file1 = sys.argv[1]
+file2 = sys.argv[2]
+
+if not (file1.endswith(".json") or file1.endswith(".csv")) and not (file2.endswith(".json") or file2.endswith(".csv")):
+    print("Provided files must either be .json files or .csv files", file=sys.stderr)
+    sys.exit(1)
+
+# read contents of files into a dataframe in preparation for comparison
+# note: we're assuming that the provided input is derived from the json output of the speed tool
+df1 = read_data(file1)
+df2 = read_data(file2)
+
+# only compare benchmarks that appear in both of the files
+# We need this because the speed tool at the time of writing has some tests that are disabled for OpenSSL
+# we're using .iloc[:, 0] here because we're filtering out rows where the content of the 0th index column in one df isn't in the other
+# details of .iloc can be found here: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.iloc.html
+# .shape[0] represents the number of rows in the dataframe
+# details of .shape can be found here: https://pandas.pydata.org/pandas-docs/version/0.23/generated/pandas.DataFrame.shape.html
+if df1.shape[0] > df2.shape[0]:
+    df1 = df1[df1.iloc[:, 0].isin(df2.iloc[:, 0])]
+elif df2.shape[0] > df1.shape[0]:
+    df2 = df2[df2.iloc[:, 0].isin(df1.iloc[:, 0])]
+
+# 2nd column of the dataframe represents the number of calls
+df1_numCalls = df1.iloc[:, 1]
+df2_numCalls = df2.iloc[:, 1]
+
+# put both dataframes side by side for comparison
+dfs = pd.concat([df1, df2], axis=1)
+
+# we want things that have a +15% regression
+compared = np.where(((df2_numCalls - df1_numCalls)/df1_numCalls*100 <= -15), df1.iloc[:, 0], np.nan)
+
+compared_df = dfs.loc[dfs.iloc[:, 0].isin(compared)]
+compared_df = compared_df.dropna()
+compared_df["Percentage Difference"] = ((compared_df.iloc[:, 5] - compared_df.iloc[:, 1])/compared_df.iloc[:, 1]*100)
+
+print(compared_df)
```

Comment:
Yes, all microsecond values will be similar enough that any variation that it introduces will be missed by the 15% margin that we are looking for. Thus we can just look at the number of calls.

Reply:
Below is not required in merging this PR.
* A comment is needed to tell why `15%` is the threshold. Is it because this is the largest gap we observed?